### ISSUE TITLE
Poisson NLL loss in libtorch

### ISSUE THREAD

**Issue by interesaaat** [pytorch/pytorch#19186]
## 🚀 Feature
There is no Poisson NLL loss in libtorch. 

## Motivation

Needed when running models with Poisson NLL loss using libtorch. 

## Pitch

I can work on it if someone advice me on which files to touch. 
Where should the implementation go? In `aten\native\Loss.cpp` plus a binding in `native_functions.yaml`? Where should tests go?

**Comment by soumith** [comment]
if it is for the C++ frontend, the implementation goes here: https://github.com/pytorch/pytorch/tree/master/torch/csrc/api/src/nn
The tests go here: https://github.com/pytorch/pytorch/tree/master/test/cpp/api

**Comment by interesaaat** [comment]
I see. So i can create a loss.cpp file and put the code there I guess. Why is supposed to be here and not in Aten? I don't see any other loss function implementation here.

**Comment by soumith** [comment]
the other losses are already implemented in aten, while poisson_nll_loss is implemented in pure python: https://github.com/pytorch/pytorch/blob/447d74a0747be7686dc44bb9db57ca45954fc865/torch/nn/functional.py#L1925-L1943

The other option is to move that loss to aten, similar to nll_loss, and then call it from python as torch._C.nn.nll_loss: https://github.com/pytorch/pytorch/blob/447d74a0747be7686dc44bb9db57ca45954fc865/torch/nn/functional.py#L1865

Then, it should just work out in the C++ API as well

**Comment by interesaaat** [comment]
Yes exactly, my point is that probably it is better to have the loss directly in Aten instead of having two implementations (one in python and one in c++).

**Comment by soumith** [comment]
yea you're right. just put it in aten/native
